flink-connector-akka/src/main/java/org/apache/flink/streaming/connectors/akka/AkkaSource.java

```diff
@@ -0,0 +1,147 @@
+  implements StoppableFunction {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AkkaSource.class);
+
+  private static final long serialVersionUID = 1L;
+
+  // --- Fields set by the constructor
+
+  private final Class<?> classForActor;
+
+  private final String actorName;
+
+  private final String urlOfPublisher;
+
+  // --- Runtime fields
+  private transient ActorSystem receiverActorSystem;
+  private transient ActorRef receiverActor;
+  private transient Object waitLock;
+  private transient boolean running = true;
+
+  protected transient boolean autoAck;
+
+  /**
+   * Creates {@link AkkaSource} for Streaming
+   *
+   * @param actorName Receiver Actor name
+   * @param urlOfPublisher tcp url of the publisher or feeder actor
+   */
+  public AkkaSource(String actorName,
+          String urlOfPublisher) {
+    super();
+    this.classForActor = ReceiverActor.class;
+    this.actorName = actorName;
+    this.urlOfPublisher = urlOfPublisher;
+  }
+
+  @Override
+  public void open(Configuration parameters) throws Exception {
+    waitLock = new Object();
+    receiverActorSystem = createDefaultActorSystem();
+
+    RuntimeContext runtimeContext = getRuntimeContext();
+    if (runtimeContext instanceof StreamingRuntimeContext
+      && ((StreamingRuntimeContext) runtimeContext).isCheckpointingEnabled()) {
+      autoAck = false;
+    } else {
+      autoAck = true;
+    }
+  }
+
+  @Override
+  public void run(SourceFunction.SourceContext<Object> ctx) throws Exception {
+    LOG.info("Starting the Receiver actor {}", actorName);
+    receiverActor = receiverActorSystem.actorOf(
+      Props.create(classForActor, ctx, urlOfPublisher, autoAck), actorName);
+
+    running = true;
+    LOG.info("Started the Receiver actor {} successfully", actorName);
+
+    while (running) {
+      synchronized (waitLock) {
+        waitLock.wait(100L);
+      }
+    }
+  }
+
+  @Override
+  public void close() {
+    this.running = false;
+    LOG.info("Closing source");
+    if (receiverActorSystem != null) {
+      receiverActor.tell(PoisonPill.getInstance(), ActorRef.noSender());
+      receiverActorSystem.shutdown();
+      receiverActorSystem.awaitTermination();
+    }
+    synchronized (waitLock) {
+      waitLock.notify();
+    }
+  }
+
+  @Override
+  public void cancel() {
+    LOG.info("Cancelling akka source");
+    close();
+  }
+
+  @Override
+  public void stop() {
+    LOG.info("Stopping akka source");
+    close();
+  }
+
+  /**
+   * Creates an actor system with default configurations for Receiver actor.
+   *
+   * @return Actor System instance with default configurations
+   */
+  private ActorSystem createDefaultActorSystem() {
+    String defaultActorSystemName = "receiver-actor-system";
+
+    String configString = "akka.actor.provider = \"akka.remote.RemoteActorRefProvider\"\n" +
+      "akka.remote.enabled-transports = [\"akka.remote.netty.tcp\"]";
+    Config defaultConfig = ConfigFactory.parseString(configString);
```

Comment:
I wonder if we should allow users of the source to add more configuration options.